dp/heights_bumper/Solution.java

```diff
@@ -48,8 +48,8 @@ public int numToRemoveToFormPike(int[] heights) {
 
 class Memoization implements Solution {
 
-    private int recursion(int[] heights, int i, int lastHeightIdx, boolean increasing, int removed, int[][][] dp) {
```

Comment:
Not that sure why passing return value as parameter slows down the runtime so much (making it even much slower than brute force recursion, as the relevant issue described). But by not passing this unnecessary variable as parameter solves it.